### PR TITLE
test: add test for getBoostedQuests

### LINKED ISSUE #606: test: `/get_boosted_quests`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by ibsule**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

Resolves: #606

### CODE REVIEW

**Review by fricoben — APPROVED** [review]
lgtm

File: tests/services/apiService.test.js [hunk 1]
```diff
@@ -40,3 +40,52 @@ describe("fetchQuestCategoryData function", () => {
...
+    });
+
+    const result = await getBoostedQuests();
+
+    expect(fetch).toHaveBeenCalledWith(`${API_URL}/get_boosted_quests`);
+    expect(result).toEqual(result);
+  });
+
+  it("should handle unexpected reesponse", async () => {
+    const mockData = {
```
Comment (ayushtom) [inline tests/services/apiService.test.js:94]: you can add this message here - `Error querying boosts`

File: tests/services/apiService.test.js [hunk 2]
```diff
@@ -40,3 +40,52 @@ describe("fetchQuestCategoryData function", () => {
...
   });
 });
+
+describe("getBoostedQuests function", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+  });
+
+  it("should fetch and return all boosted quests", async () => {
+    const mockData = [23, 104, 24, 105, 106, 26, 27];
```
Comment (ayushtom) [inline tests/services/apiService.test.js:68]: can you also add this as a type in the api call in apiService.ts

**Comment by ibsule** [comment]
Hi @ayushtom, I noticed that a couple of PRs are consistently having merged conflicts due to many people making changes to one file. I suggest that every endpoint test should have a separate file. This will avoid merged conflicts to a larger extent and make the contributions cleaner.

An example could be `getBoostedQuests.test.ts` which will contain only the tests I wrote for `get_boosted_endpoint`

**Comment by ayushtom** [comment]
hey @princeibs , i think in this particular scenario only multiple people are working on the same file, but moving forward this file will only have incremental changes hence i think for now it's okay. I dont think test cases should be split into files since they are usually sanitary checks and won't require frequent changes. Making it cleaner to navigate isn't a priority

**Comment by ayushtom** [comment]
I also resolved the conflicts you had so should be easier to work on now

**Comment by ibsule** [comment]
All noted @ayushtom, let me fix the requested change

**Comment by ibsule** [comment]
@ayushtom I think everything has been resolved. Please let me know if there is anything I need to add

**Comment by ayushtom** [comment]
@princeibs you have build errors
you can run npm run build locally to fix these and if all looks good then please push the changes so i can merge.
Test cases look good

**Comment by ibsule** [comment]
@ayushtom build error fixed✅

**Pull request merged**